proto/palomachain/paloma/evm/user_smart_contract.proto

```diff
@@ -0,0 +1,31 @@
+syntax = "proto3";
+package palomachain.paloma.evm;
+
+import "cosmos_proto/cosmos.proto";
+
+option go_package = "github.com/palomachain/paloma/x/evm/types";
+
+enum DeploymentStatus {
+  DEPLOYMENT_STATUS_UNSPECIFIED = 0;
+  DEPLOYMENT_STATUS_ONGOING = 1;
+  DEPLOYMENT_STATUS_ACTIVE = 2;
+  DEPLOYMENT_STATUS_ERROR = 3;
+}
+
+// UserSmartContract defines user-uploaded smart contracts
+// We keep them in storage, so users can decide to deploy them anywhere
+message UserSmartContract {
+  message Deployment {
+    string chain_reference_id = 1;
+    DeploymentStatus status = 2;
+    string address = 3;
+  }
+
```

Comment:
One more thing, we probably want to make sure we evict contracts from the DB after a certain amount of block height inactivity to avoid chain bloat. 

@taariq That means we'd clear the data on Paloma after `n` blocks, so if I wanted to deploy the same contract to a new chain, I'd have to register it again. To be clear, the deployments that already happened won't be touched, they stay of course. If we leave this data on chain forever, it will grow and grow and grow. My suggestion would be to evict contracts which haven't been deployed to any chains in 30 days.